src/CollectWidget.res

```diff
@@ -25,7 +31,36 @@ let make = (
   ) = React.useState(_ => None)
   let (submitted, setSubmitted) = React.useState(_ => false)
   let (paymentMethodData, setPaymentMethodData) = React.useState(_ => Dict.make())
+
+  // Input DOM references
   let inputRef = React.useRef(Nullable.null)
+  let cardNumberRef = React.useRef(Nullable.null)
+  let cardExpRef = React.useRef(Nullable.null)
+  let cardHolderRef = React.useRef(Nullable.null)
+  let routingNumberRef = React.useRef(Nullable.null)
+  let achAccNumberRef = React.useRef(Nullable.null)
+  let bacsSortCodeRef = React.useRef(Nullable.null)
+  let bacsAccNumberRef = React.useRef(Nullable.null)
+  let ibanRef = React.useRef(Nullable.null)
+  let sepaBicRef = React.useRef(Nullable.null)
+  let bankNameRef = React.useRef(Nullable.null)
+  let bankCityRef = React.useRef(Nullable.null)
+  let countryCodeRef = React.useRef(Nullable.null)
+
+  // Init
+  React.useEffect1(() => {
+    switch formLayout {
+    | Tabs => setSelectedPaymentMethodType(_ => availablePaymentMethodTypes->Array.get(0))
+    | _ => ()
+    }
+    None
+  }, [availablePaymentMethodTypes])
+
+  // Update availablePaymentMethodTypesOrdered
+  React.useEffect(() => {
+    setAvailablePaymentMethodTypesOrdered(_ => availablePaymentMethodTypes)
```

Comment:
why not move this as well to the previous useEffect if the dependency is the same